httpclient/socksify.go

```diff
@@ -18,6 +22,28 @@ func SOCKS5DialFuncFromEnvironment(origDialer DialFunc) DialFunc {
 		return origDialer
 	}
 
+	if strings.HasPrefix(allProxy, "ssh+") && strings.Contains(allProxy, "?private-key=") {
+		allProxy = strings.TrimPrefix(allProxy, "ssh+socks5://")
```

Comment:
can we use URL parse to grab all these pieces?

Reply:
Yup.